### ISSUE TITLE
Unclear file with corrupt document or bad syntax

### ISSUE THREAD

**Issue by DAB0mB** [dotansimha/graphql-code-generator#1129]
When we specify `config.documents`, given a file which contains a bad syntax or corrupt documents, its path won't be reported.

**Example 1 - Bad syntax**

Given the file `a/b/file.js`

```js
gql `
query Me {
  me {
    ...User
  }
}
${fragments.user}
`

const func()
```

We will get the error `',' expected.` without knowing at which file it occurred in. I would expect it to print the file path `a/b/file.js`

**Example 2 - Corrupt documents**

Given the file `a/b/file.js`

```js
gql `
query Me {
  me {
    ...User
  }
}
`
```

We will get the error `Cannot read property 'onType' of undefined` without knowing at which file it occurred in. I would expect it to print the file path `a/b/file.js`.

**Comment by kamilkisiela** [comment]
I couldn't reproduce exact issues you have but yes, there's no file paths next to error messages. Other than that it works.

https://github.com/dotansimha/graphql-code-generator/blob/d1e060cc3783df6c6514670482053afe5b543a9f/dev-test/test-message/documents.ts#L4-L11

<img width="1035" alt="zrzut ekranu 2019-01-15 o 12 07 39" src="https://user-images.githubusercontent.com/8167190/51176936-883c0b00-18be-11e9-983c-6206cb1a82ac.png">

https://github.com/dotansimha/graphql-code-generator/blob/1ae79418310952ee0d8388d632379fb6596e126d/dev-test/test-message/documents.ts#L10-L13

<img width="616" alt="zrzut ekranu 2019-01-15 o 12 08 06" src="https://user-images.githubusercontent.com/8167190/51176926-83775700-18be-11e9-8b37-c1dbc3e25e9c.png">

**Comment by DAB0mB** [comment]
@kamilkisiela Yes it does report which file it failed to generate, but it did't report **which source file** caused it to break. If I have a syntax error in `graphql-code-generator/dev-test/test-message/documents.ts` I wanna know that. It can be in any file. It can be extremely confusing in large projects. So in addition to:

    failed to generate ./foo/bar/types.ts

I wanna have

    because of ./foo/bar/documents.ts

**Comment by kamilkisiela** [comment]
@DAB0mB That's what I wrote :) it's still an issue, I just confirmed and added more information.

**Comment by dotansimha** [comment]
Fixed in 0.16.0
